test/factories/host_related.rb

```diff
@@ -174,8 +174,8 @@ def set_nic_attributes(host, attributes, evaluator)
       after(:create) do |host,evaluator|
         evaluator.report_count.times do |i|
           report = FactoryGirl.create(:report, :host => host, :reported_at => (evaluator.report_count - i).minutes.ago)
-          host.last_report = report.reported_at
         end
+        host.update_attribute(:last_report, host.reports.last.reported_at)
```

Comment:
Otherwise the factory generated host doesn't have the last_report attribute set correctly for some reason. also this removes the n+1 of updating the last report for each report.